lib/gherlint/GherlintConfig.js

```diff
@@ -13,72 +13,80 @@ const Fs = require("../../utils/Fs");
-    exitCode = 0;
+    #configFilePattern = null;
+    #defaultConfig = null;
 
     static get CONFIG_FILE_NAME() {
         return ".gherlintrc";
     }
 
+    static get FEATURE_FILE_EXTENSION() {
+        return ".feature";
+    }
+
     static get CONFIG_EXTENSION() {
         return Object.freeze({ json: ".json", js: ".js" });
     }
 
     constructor(cliOptions = {}) {
         this.cliOptions = cliOptions;
 
-        this.configFilePattern = configFilePattern;
-        this.defaultConfig = defaultGherlintrc;
-        this.configFile = null;
-        this.configuration = null;
-        this.files = [];
-
-        this.#setup();
+        this.#configFilePattern = configFilePattern;
+        this.#defaultConfig = defaultGherlintrc;
+        this.configFilePath = null;
+        this.configuration = {};
+        this.featureFiles = [];
     }
 
-    #setup() {
-        this.configFile = this.#getConfigFile();
-        if (this.configFile) {
-            this.configuration = this.#readConfigFromFile(this.configFile);
-        }
-        if (!this.configuration) {
-            this.configuration = this.getDefaultConfig();
-        }
+    init() {
+        let config = {};
+        this.configFilePath = this.getConfigFilePath();
 
-        this.configuration = this.#overrideByCliConfig(
-            this.configuration,
-            this.cliOptions.cliConfig
-        );
+        if (this.configFilePath) {
+            config = this.readConfigFromFile(this.configFilePath);
+            this.validateConfig(config);
+        }
 
-        // attach cliOptions to configuration
-        delete this.cliOptions.cliConfig;
-        this.configuration.cliOptions = this.cliOptions;
+        config = this.mergeWithDefaultConfig(config);
 
-        this.files = this.#listFeatureFiles(this.configuration.files);
-    }
+        // override configuration by cli options
+        this.configuration = this.overrideByCliConfig(config, this.cliOptions);
 
-    #hasCliConfigFileOption() {
-        return this.cliOptions.config ? true : false;
-    }
+        this.featureFiles = this.getFeatureFiles(this.configuration.files);
 
-    #getConfigFile() {
-        if (this.configFile) return this.configFile;
+        // Todo: implement ignorePatterns
 
-        if (this.#hasCliConfigFileOption()) {
-            if (isDir(this.cliOptions.config)) {
-                log.error(
-                    "'-c, --config' option takes file only.",
-                    "Usage:",
-                    "gherlint -c path/to/.gherlintrc"
-                );
-            }
+        // remove properties related to files
+        delete this.configuration.files;
+        delete this.configuration.ignorePatterns;
+    }
 
-            return this.cliOptions.config;
-        } else {
-            return this.#searchConfigFile();
-        }
+    // Todo: implement --config cli option
+    // #hasCliConfigFileOption() {
+    //     return this.cliOptions.config ? true : false;
+    // }
```

Comment:
Do we need this?

Reply:
those are ancient code which I disabled with new structure/architecture. We want to implement them soon